src/components/GamepadController.js

```diff
@@ -0,0 +1,33 @@
+import React, { useContext, useEffect, useState } from 'react';
+
+import Fab from '@material-ui/core/Fab';
+import Gamepad from '@material-ui/icons/Gamepad';
+
+import { Context as ControllerContext } from '../context/Controller';
+
+export default function GamepadComponent() {
+    const [connected, setConnected] = useState(false)
+    useEffect(() => {
+        window.addEventListener('gamepadconnected', function(e) {
+            console.log('gamepadconnected');
+            setConnected(true); 
+        });
+        window.addEventListener('gamepaddisconnected', function(e) {
+            console.log('gamepadconnected');
+            setConnected(false);
+        });
+
+        return function cleanup() {
+            
+        };
```

Comment:
Why is there an empty cleanup function here?
Aren't you supposed to remove events when unmounted?

Reply:
will do later when finalizing gamepad controller

Reply:
Why not in this PR?

Reply:
On it!

Reply:
Done.